core/embed/rust/src/ui/model_tr/layout.rs

```diff
@@ -392,6 +412,19 @@ extern "C" fn new_confirm_properties(n_args: usize, args: *const Obj, kwargs: *m
     unsafe { util::try_with_args_and_kwargs(n_args, args, kwargs, block) }
 }
 
+extern "C" fn new_confirm_homescreen(n_args: usize, args: *const Obj, kwargs: *mut Map) -> Obj {
+    let block = move |_args: &[Obj], kwargs: &Map| {
+        let title: StrBuffer = kwargs.get(Qstr::MP_QSTR_title)?.try_into()?;
+        let data: Obj = kwargs.get(Qstr::MP_QSTR_image)?;
+
+        let buffer = unsafe { unwrap!(get_buffer(data)) };
```

Comment:
You can't keep a reference to the buffer and return to micropython. This can be worked around with closures, see `new_confirm_homescreen` in `ui/model_tt/layout.rs`.

Reply:
Right, changed in [06a3e8b](https://github.com/trezor/trezor-firmware/pull/3287/commits/06a3e8bdb4e93732942ef533e4e68fbda42e5ce9)

What things could happen when we would not do this?

Reply:
Due to how buffers work in micropython, the garbage collector would not understand that we have a reference to the buffer and might deallocate it while in use by rust code.

Also `(b)` is violated here because we cannot guarantee that python code does not touch the buffer once we return to micropython:
> SAFETY: The caller is responsible for ensuring immutability of the returned buffer, in particular that: ... (b) the buffer is not modified in MicroPython while the reference to it is being held.